cmake/modules/AddSwift.cmake

```diff
@@ -435,8 +435,9 @@ function(add_swift_host_library name)
   endif()
 
   if(XCODE)
-    get_filename_component(dir ${CMAKE_CURRENT_SOURCE_DIR} DIRECTORY)
-
+    string(REGEX MATCHALL "/[^/]+" split_path ${CMAKE_CURRENT_SOURCE_DIR})
+    list(GET split_path -1 dir)
```

Comment:
Hmm, I'm not sure I understand how this differs.  Is this not getting the basename of the current source dir?

Reply:
Yes, but the wrong code did get the directory.
Sorry, I didn't spent too much time on it - I just reverted.